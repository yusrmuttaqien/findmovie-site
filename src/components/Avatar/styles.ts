import styled from 'styled-components';

export const Wrapper = styled.figure`
  --_dimensions: 2.5rem;
  background-color: red;
  height: var(--_dimensions);
  width: var(--_dimensions);
  border-radius: 50%;
  overflow: hidden;

  img {
    height: var(--_dimensions);
    width: var(--_dimensions);
  }

  @media screen and (min-width: ${({ theme }) => theme.screen.desktop}px) {
    --_dimensions: 4rem;
  }
`;
