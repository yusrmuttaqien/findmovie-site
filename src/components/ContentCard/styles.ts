import { Link as L } from 'react-router-dom';
import styled from 'styled-components';

interface BackdropProps {
  readonly $placeholder: string[];
}

export const Wrapper = styled.figure`
  padding: 0.75rem 0.8125rem;
  position: relative;
  overflow: hidden;

  #contents {
    position: relative;
    z-index: 2;
    margin-top: 3.375rem;

    #content-subdetail {
      font-size: 0.75rem;
    }

    #content-detail {
      font-size: 0.85rem;
      font-weight: 200;
      display: -webkit-box;
      overflow: hidden;
      text-overflow: ellipsis;
      -webkit-box-orient: vertical;
      -webkit-line-clamp: 4;
    }
  }

  &:hover {
    #backdrop-image {
      transform: scale(1.209);
    }
  }

  @media screen and (min-width: ${({ theme }) => theme.screen.desktop}px) {
    padding: 1.5rem 1.6875rem;

    #contents {
      margin-top: 0;

      #content-detail {
        font-size: 0.9375rem;
        -webkit-line-clamp: 3;
        margin-right: 35%;
      }
    }
  }
`;

export const Link = styled(L)`
  font-size: 1.25rem;
  text-decoration: underline;
  font-weight: 600;
  margin-bottom: 0.5rem;
  display: block;
  display: -webkit-box;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 1;
  cursor: ${({ to }) => (to === '#' ? 'default' : 'pointer')};

  @media screen and (min-width: ${({ theme }) => theme.screen.desktop}px) {
    font-size: 1.875rem;
  }
`;

export const Backdrop = styled.div<BackdropProps>`
  z-index: 1;
  position: absolute;
  top: -10px;
  bottom: -10px;
  left: -10px;
  right: -10px;
  background-image: ${({ $placeholder }) => `url(${$placeholder[1]})`};
  background-size: cover;
  background-position: top right;

  img {
    transform: scale(1.1);
    transition: transform 0.3s ease-in-out;
    width: 100%;
    height: 100%;
    object-fit: cover;
  }

  &::after {
    content: '';
    position: absolute;
    inset: 0;
    background: linear-gradient(
      360deg,
      rgba(0, 0, 0, 0.91) 0%,
      rgba(0, 0, 0, 0.85) 26.14%,
      rgba(0, 0, 0, 0.71) 46.06%,
      rgba(0, 0, 0, 0.4) 70.73%,
      rgba(217, 217, 217, 0) 100%
    );
    backdrop-filter: blur(2px);
  }

  @media screen and (min-width: ${({ theme }) => theme.screen.desktop}px) {
    background-image: ${({ $placeholder }) => `url(${$placeholder[0]})`};
    background-position: right center;

    &::after {
      background: linear-gradient(
        90deg,
        rgba(0, 0, 0, 0.91) 0%,
        rgba(0, 0, 0, 0.85) 25%,
        rgba(0, 0, 0, 0.71) 50%,
        rgba(0, 0, 0, 0.4) 75%,
        rgba(217, 217, 217, 0) 100%
      );
    }
  }
`;
