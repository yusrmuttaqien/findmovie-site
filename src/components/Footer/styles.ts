import styled from 'styled-components';

export const Wrapper = styled.footer`
  padding: 1rem 0;
  border-top: 1px solid;
  border-color: ${({ theme }) => theme.colors.white[50]};
  margin-top: 1.25rem;

  p {
    font-size: 0.75rem;
    font-weight: 200;
    text-align: center;
    color: ${({ theme }) => theme.colors.white[70]};
  }

  @media screen and (min-width: ${({ theme }) => theme.screen.desktop}px) {
    margin-top: 2.1875rem;

    p {
      font-size: 1rem;
    }
  }
`;
