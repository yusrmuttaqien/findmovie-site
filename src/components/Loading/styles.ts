import styled from 'styled-components';

export const Wrapper = styled.div`
  overflow: hidden;
  width: 100%;
  height: max-content;

  span {
    display: block;
    width: 100%;
    height: 4px;
    background-color: ${({ theme }) => theme.colors.white[100]};
    animation: stripAnimation 1s infinite linear;
  }

  @keyframes stripAnimation {
    0% {
      transform: translateX(-100%);
    }
    100% {
      transform: translateX(100%);
    }
  }
`;
