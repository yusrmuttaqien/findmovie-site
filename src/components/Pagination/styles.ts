import styled from 'styled-components';

interface ButtonProps {
  readonly $isActive?: boolean;
}

export const Wrapper = styled.div`
  display: flex;
  justify-content: space-between;
`;

export const Button = styled.button<ButtonProps>`
  color: ${({ theme }) => theme.colors.white[50]};
  background-color: transparent;
  border: none;
  cursor: pointer;

  p,
  svg {
    transition: color 0.1s ease-in-out;
  }

  p {
    font-size: 1.0625rem;
  }

  svg {
    font-size: 1.25rem;
  }

  &:last-child {
    flex-direction: row-reverse;

    svg {
      transform: rotate(180deg);
    }
  }

  &:hover {
    p,
    svg {
      color: ${({ theme }) => theme.colors.white[100]};
    }
  }

  &:disabled {
    cursor: no-drop;

    p,
    svg {
      color: ${({ theme, $isActive }) => {
        return $isActive ? theme.colors.white[100] : theme.colors.white[30];
      }};
    }
  }

  @media screen and (min-width: ${({ theme }) => theme.screen.desktop}px) {
    p {
      font-size: 1.75rem;
    }

    svg {
      font-size: 2rem;
    }
  }
`;
