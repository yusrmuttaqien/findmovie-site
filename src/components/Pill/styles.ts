import styled from 'styled-components';

export const Wrapper = styled.figure`
  background-color: ${({ theme }) => theme.colors.white[100]};
  padding: 0.25rem 0.375rem;
  border-radius: 0.25rem;

  p {
    color: ${({ theme }) => theme.colors.black[100]};
    font-size: 0.625rem;
    font-weight: 500;
  }

  @media screen and (min-width: ${({ theme }) => theme.screen.desktop}px) {
    padding: 0.375rem 0.5rem;

    p {
      color: ${({ theme }) => theme.colors.black[100]};
      font-size: 1rem;
    }
  }
`;
