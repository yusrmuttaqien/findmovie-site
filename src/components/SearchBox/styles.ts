import styled from 'styled-components';
import { SearchIcon as SI } from 'assets/svgs';

export const Wrapper = styled.div`
  display: flex;
  align-items: center;
  width: max-content;
  gap: 1.25rem;

  input {
    width: 11.875rem;
    background-color: transparent;
    border: none;
    color: ${({ theme }) => theme.colors.white[100]};
    font-size: 1.0625rem;
    line-height: 1.5;

    &::placeholder {
      color: ${({ theme }) => theme.colors.white[50]};
    }

    @media screen and (min-width: ${({ theme }) => theme.screen.desktop}px) {
      font-size: 1.3rem;
      width: 15rem;
    }
  }
`;

export const SearchIcon = styled(SI)`
  font-size: 1.25rem;
  cursor: pointer;

  @media screen and (min-width: ${({ theme }) => theme.screen.desktop}px) {
    font-size: 2rem;
  }
`;
