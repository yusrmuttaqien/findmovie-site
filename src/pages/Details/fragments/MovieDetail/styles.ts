import styled from 'styled-components';

export const Wrapper = styled.section`
  display: flex;
  flex-direction: column;
  gap: 1.5625rem;
  padding: 0 ${({ theme }) => theme.space.x.mobile};

  #metadatas {
    h3 {
      font-weight: 500;
      font-size: 1rem;

      &:not([data-meta='title']) {
        margin-bottom: 0.5rem;
      }

      span {
        font-weight: 800;
      }
    }

    > div {
      display: flex;
      flex-wrap: wrap;
      gap: 0.375rem;
    }

    &:last-child p {
      font-size: 0.85rem;
      font-weight: 200;
    }
  }

  @media screen and (min-width: ${({ theme }) => theme.screen.desktop}px) {
    gap: 2rem;
    padding: 0 ${({ theme }) => theme.space.x.desktop};

    #metadatas {
      h3 {
        font-size: 2.125rem;

        &:not([data-meta='title']) {
          margin-bottom: 1rem;
        }
      }

      > div {
        gap: 0.625rem;
      }

      &:last-child p {
        font-size: 1.5rem;
      }
    }
  }
`;
