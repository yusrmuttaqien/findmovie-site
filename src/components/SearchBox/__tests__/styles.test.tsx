import { describe, test, expect } from 'vitest';
import { render } from '@testing-library/react';
import { Wrapper, SearchIcon } from '../styles';
import theme from 'styles/index';
import 'jest-styled-components';

describe('SearchBox component styles', () => {
  test('styled components should match snapshot', () => {
    const { asFragment: fWrapper } = render(<Wrapper theme={theme} />);
    const { asFragment: fSearchIcon } = render(<SearchIcon theme={theme} />);

    expect(fWrapper()).toMatchSnapshot();
    expect(fSearchIcon()).toMatchSnapshot();
  });
});
