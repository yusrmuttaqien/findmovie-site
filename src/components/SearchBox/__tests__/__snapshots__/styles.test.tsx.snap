// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchBox component styles > styled components should match snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: flex;
  align-items: center;
  width: max-content;
  gap: 1.25rem;
}

.c0 input {
  width: 11.875rem;
  background-color: transparent;
  border: none;
  color: rgb(199, 199, 199);
  font-size: 1.0625rem;
  line-height: 1.5;
}

.c0 input::placeholder {
  color: rgba(199, 199, 199, 0.5);
}

@media screen and (min-width: 1279px) {
  .c0 input {
    font-size: 1.3rem;
    width: 15rem;
  }
}

@media screen and (min-width: 1279px) {

}

<div
    class="c0"
  />
</DocumentFragment>
`;

exports[`SearchBox component styles > styled components should match snapshot 2`] = `
<DocumentFragment>
  .c0 {
  font-size: 1.25rem;
  cursor: pointer;
}

@media screen and (min-width: 1279px) {

}

@media screen and (min-width: 1279px) {
  .c0 {
    font-size: 2rem;
  }
}

<svg
    class="c0"
    fill="none"
    height="1em"
    viewBox="0 0 21 21"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M17.6153 17.9231L14.0519 14.3596M14.0519 14.3596C14.6614 13.75 15.1449 13.0264 15.4748 12.23C15.8047 11.4336 15.9745 10.58 15.9745 9.71796C15.9745 8.85593 15.8047 8.00233 15.4748 7.20592C15.1449 6.40951 14.6614 5.68587 14.0519 5.07632C13.4423 4.46677 12.7187 3.98325 11.9223 3.65336C11.1258 3.32348 10.2723 3.15369 9.41022 3.15369C8.54819 3.15369 7.6946 3.32348 6.89818 3.65336C6.10177 3.98325 5.37813 4.46677 4.76858 5.07632C3.53754 6.30736 2.84595 7.97701 2.84595 9.71796C2.84595 11.4589 3.53754 13.1286 4.76858 14.3596C5.99962 15.5906 7.66927 16.2822 9.41022 16.2822C11.1512 16.2822 12.8208 15.5906 14.0519 14.3596Z"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="0.820513"
    />
  </svg>
</DocumentFragment>
`;
